packages/core/src/components/essentials/title.ts

```diff
@@ -89,12 +94,15 @@ export class Title extends Component {
 	}
 
 	// computes the maximum space a title can take
-	protected getMaxTitleWidth() {
+	protected getMaxTitleWidth(toolbarEnabled) {
 		const containerWidth = DOMUtils.getSVGElementSize(
 			this.services.domUtils.getMainSVG(),
 			{ useAttr: true }
 		).width;
-		return containerWidth;
+		const finalContainerWidth = toolbarEnabled
+			? containerWidth - Configuration.toolBar.iconSize * 3
+			: containerWidth;
+		return finalContainerWidth;
```

Comment:
toolbar should really be positioned using `Layout` components at the `AxisChart` level, rather than adding hacks here, because when we add more icons to the toolbar, then we'd need to change the number `3`

Reply:
@theiliad 
Updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/a734ef0502eede584e05dc56519e96f2f9a92a00.

Please confirm that this change meets your layout expectation.
I will refactor more ToolBar code following this approach.
Thanks.